### PR TITLE
Chrished/issue52

### PR DESCRIPTION
**Pull request by chrished**
1. Create table with language of title
2. add extra columns to author output restricting papers to english language/paper being the main paper of a family

### CODE REVIEW

File: src/dataprep/main/prep_mag/paper_language.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+            # Put results in the queue
+            for batch_result in results:
+                db_queue.put(batch_result)
+
+            # Print progress
+            processed_papers = sum(len(batch) for batch in results)
+            progress = processed_papers / total_papers * 100
+            elapsed_time = time.time() - start_time
+            print(f"Progress: {progress:.2f}% ({processed_papers}/{total_papers}) - "
```
Comment (f-hafner) [inline src/dataprep/main/prep_mag/paper_language.py:138]: I think this prints only at the end of processes, in which I'd suggest to remove it

**Review by f-hafner — CHANGES_REQUESTED** [review]
Thanks, @chrished !
I went through the multiprocessing in a bit more detail to understand what's going on :) I added some docstrings, moved the `parse_args` function to the bottom, just before the `main` call, and raise a `RuntimeError` instead of printing a warning when not all papers are present in the new table.

I like the idea of writing the results back in the same process; what I've done so far was to write out the result of the child processes into csv files, and then reading them back in separately. 

Here are some comments:
- why is the `Manager` necessary? If I understand correctly, there is also a `multiprocessing.Queue`, which I think would suffice in this context. (see https://docs.python.org/3/library/multiprocessing.html#pipes-and-queues)
- the current way to query papers in the children assumes that `select` statements on sqlite always return results in the same order. But this is in fact not guaranteed by sqlite -- and could introduce a subtle bug that will be hard to track down because it's nothing that is wrong per se in the code. I think it'd be better to be more explicity and pass the `PaperId`s to query to each child process.
- you currently create only as many chunks as you have cores on the machine. an alternative would be to fix the chunk size and then determine the number of processes; if there are more processes than cores, they won't all start at the same time. For me it's fine to leave it like this, but maybe we can add a note/warning somewhere to not forget.

File: src/dataprep/main/prep_mag/paper_language.py (hunk 1 above)
Comment (chrished) [reply src/dataprep/main/prep_mag/paper_language.py:163]: ok!

File: src/dataprep/main/prep_mag/paper_language.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import time 
+import pandas as pd
+import argparse
+from helpers.functions import print_elapsed_time, analyze_db
+from helpers.variables import db_file, insert_questionmark_doctypes, keep_doctypes
+from multiprocessing import Pool, cpu_count, Manager, Process
+from queue import Empty
+
+# Download the model if it doesn't exist
+model_path = os.path.join(os.path.dirname(__file__), 'lid.176.ftz')
```
[the remark this answers is not in the document]
Comment (chrished) [reply src/dataprep/main/prep_mag/paper_language.py:27]: ok

**Comment by chrished** [comment]
@f-hafner a few updates:

- load paperids and distribute to processes
- progress bar now measuring when a chunk finishes, before it was just measuring the starting time of all chunks (i.e. how long does it take to start all chunks).
- download fasttext model into data folder
- drop Manager and just use Queue

**Review by f-hafner — APPROVED** [review]

**Pull request merged**